syntax="proto3";


service NVM{
  rpc GetFeature(Point) returns (Feature) {}
  rpc GetDeploySrc(DeployRequest) returns (DeployResponse){}
}

message Feature{
  string name=1;
  Point location=2;
}

message Point{
  int32 latitude=1;
  int32 longitude=2;
}

message DeployRequest{
  string script=1;
  string function=2;
}

message DeployResponse{
  string result=1;
  string err_msg=2;
}
